### ISSUE TITLE
Contours in space

### ISSUE THREAD

**Issue by effigies** [poldracklab/fmriprep#1152]
https://s3.amazonaws.com/openneuro.outputs/921294bd5b869b1852ab3ce886583795/4dd151e3-52d1-4fa2-9591-27c16520331c/fmriprep/sub-001.html#Functional

See top-right of **ROIs in BOLD space**:

![bold1](https://user-images.githubusercontent.com/83442/40788307-b9958cea-64bd-11e8-91dd-7c3f8092d350.png)

**Comment by oesteban** [comment]
Yep, I've been seeing those artifacts more often. Something must have changed in nilearn proper.

**Comment by chrisfilo** [comment]
Don't we freeze the version of nilearn in the containers?

Best,
Chris

**Comment by effigies** [comment]
Their last release was March 12.

**Comment by effigies** [comment]
@chrisfilo No. We pin a specific scikit-learn, but not nilearn.

**Comment by oesteban** [comment]
March 12 seems like a probable date for these changes. I also have seen this in mriqc plots.

**Comment by effigies** [comment]
If we want to try to reproduce with nilearn==0.4.0, the above was `ds000146/sub-001`.

https://openneuro.org/datasets/ds000164/versions/00001?app=FMRIPREP&version=61&job=5b0d753f2afa3badacec83df

**Comment by SRSteinkamp** [comment]
Just to understand this from a users' perspective: 
Are these visual artifacts (e.g. some issues in the plotting function) or are these artifacts indicative that something went wrong in the ROI creation / normaliztion etc.?

**Comment by effigies** [comment]
Hi @SRSteinkamp, good question. Our understanding is that it's an issue with the plotting function (we've looked at some actual masks, and there's no indication of non-zero values there). In any event, we plan to look more closely at it and resolve the issue, whether there's a plotting problem in the upstream library or something subtly wrong with our usage.

**Comment by mpcoll** [comment]
Hi, In case this helps, I also get this for all my participants for BOLD, T1 and field maps. The actual images seem to be fine upon visual inspection. I did not notice this when I was using a previous version of fmriprep but I can't remember which one. 

I'm currently using version 1.1.1 in a docker container. 

![sub-001_t1w_seg_brainmask](https://user-images.githubusercontent.com/28481638/41791198-f2ffb8a2-764c-11e8-882c-45780a3166bb.png)
![sub-001_phasediff_fmap_mask](https://user-images.githubusercontent.com/28481638/41791202-f91d7f8a-764c-11e8-9b54-2a117a9b1eb6.png)
![sub-001_task-mvpasogl_run-1_bold_rois](https://user-images.githubusercontent.com/28481638/41791221-07c03f82-764d-11e8-8bd9-f7216c247076.png)

**Comment by oesteban** [comment]
regardless the nilearn version, this could be easily fixed improving the bounding box calculation to exclude slices with few pixels in the mask.

For instance, in the latest examples from @mpcoll, we can see how the first row is mostly brain stem in the two first cuts, and empty or almost empty in the last cut. In other words, those three cuts are almost useless. Surprisingly, for the coronal view (last row) of the fieldmap magnitude and bold registration, it seems that the bounding box was maybe too tight.
